core/bulkchange.class.inc.php

```diff
@@ -567,12 +640,94 @@ protected function PrepareObject(&$oTargetObj, $aRowData, &$aErrors)
 		}
 		return $aResults;
 	}
-	
+
+	/**
+	 * @param $sTargetClass: target class of the failed search
+	 * @param $sForeignAttCode : foreign attribute code of the foreign search
+	 * @param $value : unfound value
+	 *
+	 * @return \CellStatus_SearchIssue
+	 * @throws \CoreException
+	 * @throws \MissingQueryArgument
+	 * @throws \MySQLException
+	 * @throws \MySQLHasGoneAwayException
+	 */
+	protected function GetCellSearchIssue($sTargetClass, $sForeignAttCode, $value) : CellStatus_SearchIssue {
+		//current search with current permissions did not match
+		//let's search why and give some more feedbackst to the user
+
+		//count all objects with all permissions
+		$oReconFilter2 = new DBObjectSearch($sTargetClass);
+		$oReconFilter2->AllowAllData(true);
+		$oExtObjectSet = new CMDBObjectSet($oReconFilter2);
+		$iAllowAllDataObjectCount = $oExtObjectSet->Count();
+
+		//count all objects with current user permissions
+		$oReconFilter2->AllowAllData(false);
+		$oExtObjectSetWithCurrentUserPermissions = new CMDBObjectSet($oReconFilter2);
+		$iCurrentUserRightsObjectCount = $oExtObjectSetWithCurrentUserPermissions->Count();
+
+		if ($iAllowAllDataObjectCount === 0) {
+			//no objects at all
+			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-NoObject', $sTargetClass);
+			return new CellStatus_SearchIssue($sReason);
+
+			//search link for exhaustive values
+		}
+
+		if ($iCurrentUserRightsObjectCount === 0){
+			//no objects visible by current user
+			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-NoObject-ForCurrentUser',
+				$sTargetClass);
+			return  new CellStatus_SearchIssue($sReason);
+
+			//search link for exhaustive values
+		}
+
+		try{
+			$aDisplayedAllowedValues=[];
+			$allowedValues="";
+			$oExtObjectSetWithCurrentUserPermissions->SetLimit(4);
+			for($i=0; $i<3; $i++){
+				/** @var \DBObject $oVisibleObject */
+				$oVisibleObject = $oExtObjectSetWithCurrentUserPermissions->Fetch();
+				if (is_null($oVisibleObject)){
+					break;
+				}
+
+				$aDisplayedAllowedValues[] = $oVisibleObject->Get($sForeignAttCode);
+			}
+			$allowedValues = implode(", ", $aDisplayedAllowedValues);
+			if ($oExtObjectSetWithCurrentUserPermissions->Count()>3){
+				$allowedValues .= "...";
+			}
+		} catch(Exception $e) {
+			IssueLog::Error("failure when fetching few visible objects: " . $e->getMessage());
+		}
+
+		if ($iAllowAllDataObjectCount != $iCurrentUserRightsObjectCount) {
+			//no match and some objects NOT visible by current user. including current search maybe...
+			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-SomeObjectNotVisibleForCurrentUser',
+				$sTargetClass);
```

Comment:
```suggestion
			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-SomeObjectNotVisibleForCurrentUser', $sTargetClass);
```

No wrap below 240 chars